Feat/auth page docs (#2368)
pankod/refine: documentation/docs/core/providers/auth-provider.md

<br />

### `register`

**refine** expects this method to return a resolved Promise if the register is successful, and a rejected Promise if it is not.

> -   If the register is successfull new user is created.

Reviewer: ```suggestion
-   If the register is a successful new user is created.
```

-   If the register fails, **refine** displays an error notification to the user.

<br />

Here we show an example `register` method.

```tsx title="auth-provider.ts"
const authProvider = {
    register: ({ email, password }) => {
        // We suppose we actually send a request to the back end here.
        if (email && password) {
            // We can create a new user here.
            return Promise.resolve();
        }
        return Promise.reject();
    },
};
```

<br />

`register` method will be accessible via [`useRegister`](/core/hooks/auth/useRegister.md) auth hook.

```tsx
import { useRegister } from "@pankod/refine-core";

const { mutate: register } = useRegister();

register(values);
```

:::tip
`mutate` acquired from `useRegister` can accept any kind of object for values since `register` method from `authProvider` does not have a restriction on its parameters.  
A type parameter for the values can be provided to `useRegister`.

```tsx
const { mutate: register } =
    useRegister<{ username: string; password: string; remember: boolean }>();
```

:::

:::tip

> **refine** automatically displays an error notification if the register fails. You can customize the default error message.

Reviewer: ```suggestion
**refine** automatically displays an error notification if the registration fails. You can customize the default error message.
```

```tsx
register: ({ username, password, remember }) => {
     if (email && password) {
            return Promise.resolve();
        }
        //highlight-start
        return Promise.reject({
            name: "Register Failed!",
            message: "The username or password missing.",
        });
        //highlight-end
    },

```

:::

> [Refer to useRegister documentation for more information. &#8594](/core/hooks/auth/useRegister.md)

### `resetPassword`

**refine** expects this method to return a resolved Promise if the reset password is successful, and a rejected Promise if it is not.

> -   If the reset password is successfull you can send an email to the user with a link to reset the password.

Reviewer: ```suggestion
-   If the reset password is successful you can send an email to the user with a link to reset the password.
```
